### DISCUSSION TITLE
how to set up sshfs home file systems and login from or under another linux server

### DISCUSSION THREAD

**Discussion by christophey2k** [ltsp/ltsp#317] [Q&A]
Hello !

Have 1x  Principal Linux Server with all users and home folders and network shares (for samba user under ms windows)

Have 1x LTSP server for LTSP users ( fat disk-less computer)

How to setup system for LTSP client can use Authentication Login from my Principal Linux Server (with associated /home folders) with SSH system.

Thank you.

**Comment by alkisg** [comment]
For the accounts, you need to copy /etc/passwd and /etc/group from your SSHFS server to your LTSP server and put them in /etc/ltsp/passwd and /etc/ltsp/group.
For the SSH keys, you need to copy /etc/ssh/*.pub from your SSHFS server to your LTSP server and put them in /etc/ltsp/initrd/etc/ssh/.
To declare the different server, put the following in ltsp.conf:

```shell
[clients]
SERVER=sshfs-server-ip
```

Finally run `sudo ltsp initrd` and reboot your clients.

**Comment by christophey2k** [comment]
Thanks lot for you're fast reply ! i try but ...

On screen boot of ubuntu ltsp client,  have the good listing of usernames but that always says --> .Login incorrect
when i enter the (good) password.

Maybe because my ssh server in an Gentoo Linux machine and the mapping number of users and groups are not the same than Ubuntu LTSP server ?

Before recent day, i use an older ltsp version 5, have this configuration with success :
 the older command **# ltsp-update-sshkeys** (consist to copy ssh keys from remote auth ssh server to local ltsp server in /opt/ltsp/amd64/etc/ssh/ssh_known_hosts)
and in /etc/ltsp.conf the directive: server 192.168.253.3 (the ip of my remote ssh server who contain home folder of my users)

My users can enter username and password and have own personalized desktop, and haven't need to maintain synchronous two /etc/group and /etc/passwd files... ( the LTSP paradise ! lol )

I hope you have an idea, thank you.

**Comment by alkisg** [comment]
At the end of the [ltsp.conf man page](https://ltsp.org/man/ltsp.conf/) there's a section that says how to get a root shell on the LTSP clients.

Get a root shell, and test the following commands. Substitute "userX" with a gentoo user. What is the output of all these?

```shell
id userX
cat /etc/ltsp/ssh_known_hosts
ssh -o UserKnownHostsFile="/etc/ltsp/ssh_known_hosts" userX@server
login userX
```
